libraries/AP_Mount/AP_Mount_Backend.cpp

```diff
@@ -618,22 +618,33 @@ void AP_Mount_Backend::set_rctargeting_on_rcinput_change()
     const int16_t pitch_in = (pitch_ch == nullptr) ? 0 : pitch_ch->get_radio_in();
     const int16_t yaw_in = (yaw_ch == nullptr) ? 0 : yaw_ch->get_radio_in();
 
-    // if not in RC_TARGETING or RETRACT modes then check for RC change
     if (get_mode() != MAV_MOUNT_MODE_RC_TARGETING && get_mode() != MAV_MOUNT_MODE_RETRACT) {
-        // get dead zones
-        const int16_t roll_dz = (roll_ch == nullptr) ? 10 : MAX(roll_ch->get_dead_zone(), 10);
-        const int16_t pitch_dz = (pitch_ch == nullptr) ? 10 : MAX(pitch_ch->get_dead_zone(), 10);
-        const int16_t yaw_dz = (yaw_ch == nullptr) ? 10 : MAX(yaw_ch->get_dead_zone(), 10);
-
-        // check if RC input has changed by more than the dead zone
-        if ((abs(last_rc_input.roll_in - roll_in) > roll_dz) ||
-            (abs(last_rc_input.pitch_in - pitch_in) > pitch_dz) ||
-            (abs(last_rc_input.yaw_in - yaw_in) > yaw_dz)) {
-            set_mode(MAV_MOUNT_MODE_RC_TARGETING);
+        // if not in RC_TARGETING or RETRACT modes then check for RC change
+
+        if(last_rc_input.initialised)
+        {
+            // get dead zones
+            const int16_t roll_dz = (roll_ch == nullptr) ? 10 : MAX(roll_ch->get_dead_zone(), 10);
+            const int16_t pitch_dz = (pitch_ch == nullptr) ? 10 : MAX(pitch_ch->get_dead_zone(), 10);
+            const int16_t yaw_dz = (yaw_ch == nullptr) ? 10 : MAX(yaw_ch->get_dead_zone(), 10);
+            // check if RC input has changed by more than the dead zone
+            if ((abs(last_rc_input.roll_in - roll_in) > roll_dz) ||
+                (abs(last_rc_input.pitch_in - pitch_in) > pitch_dz) ||
+                (abs(last_rc_input.yaw_in - yaw_in) > yaw_dz)) {
+                    set_mode(MAV_MOUNT_MODE_RC_TARGETING);
+                }
+        }
+        else
+        {
+            // The first time through, initial RC inputs should be set, but not used
```

Comment:
Thanks very much for this.  Sorry to be annoying but I'd prefer if we moved this initialisation up to around line 621.  The RC initialisation doesn't need to be within the mode check I think.

I'm happy to blot over your change if you'd like.

Thanks again and sorry to be so picky.

Reply:
It's fine, it's a little bit of a brain bender to figure out what should be happening. It makes me feel uneasy somehow, but I'll try it.

Reply:
Ok I made the change, I have built it but not tested it yet, but thought I'd get you to give it a once over first. Note that the "else" at 630 is required to make this work correctly.

Reply:
Txs, just 2 more small changes.